versioning/lockfiles/configurations.rst

```diff
@@ -0,0 +1,191 @@
+
+Now, we will have 2 lockfiles, *locks/pkgb_deps_debug.lock* and *locks/pkgb_deps_release.lock*. Each one will lock different profiles and different package-id
+of ``pkga/0.1@user/testing``.
+
+
+Locked configuration
+--------------------
+
+The lockfiles store the effective configuration, settings, options, resulting from the used profiles and command line arguments.
+That configuration arguments can be passed to the ``conan lock create`` command, but not when using lockfiles. For example:
+
+.. code-block:: bash
+
+    $ mkdir build && cd build
+    $ conan install .. --lockfile=../locks/pkgb_deps_debug.lock -s build_type=Debug
+    ERROR: Cannot use profile, settings, options or env 'host' when using lockfile
+
+results in an error, because the *locks/pkgb_deps_debug.lock* already stores the ``settings.build_type`` and passing it in the command line
+could only result in inconsistencies and errors.
+
+.. important::
+
+    Lockfiles store the full effective profile configuration. It is not possible to pass configuration, settings, options or 
+    profile arguments when using lockfiles (only when creating the lockfiles)
+
+With the two captured lockfiles, now we can locally build and run our ``pkgb`` application for both configurations, guaranteeing
+the dependency to ``pkga/0.1@user/testing``:
+
+.. code-block:: bash
+
+    $ conan install .. --lockfile=../locks/pkgb_deps_release.lock
+    $ cmake ../src -G "Visual Studio 15 Win64"
+    $ cmake --build . --config Release
+    $ ./bin/greet
+    HelloA 0.1 Release
+    HelloB Release!
+    Greetings Release!
+    $ conan install .. --lockfile=../locks/pkgb_deps_debug.lock
+    $ cmake --build . --config Debug
+    $ ./bin/greet
+    HelloA 0.1 Debug
+    HelloB Debug!
+    Greetings Debug!
+
+We can again create the ``pkgb`` package for both configurations:
+
+.. code-block:: bash
+
+    $ cd ..
+    $ conan create . user/testing --lockfile=locks/pkgb_deps_release.lock --lockfile-out=locks/pkgb_release.lock
+    $ conan create . user/testing --lockfile=locks/pkgb_deps_debug.lock --lockfile-out=locks/pkgb_debug.lock
+
+
+And we could still use the lockfiles later in time to install the ``pkgb`` package with the same dependencies
+and configuration that were used to create that package:
+
+
+.. code-block:: bash
+
+    $ cd ..
+    $ mkdir consume
+    $ cd consume
+    $ conan install pkgb/0.1@user/testing --lockfile=../pkgb/locks/pkgb_release.lock
+    $ ./bin/greet
+    HelloA 0.1 Release
+    HelloB Release!
+    Greetings Release!
+    $ conan install pkgb/0.1@user/testing --lockfile=../pkgb/locks/pkgb_debug.lock
+    $ ./bin/greet
+    HelloA 0.1 Debug
+    HelloB Debug!
+    Greetings Debug!
+
+The immutability principle still holds. If we try to use *pkgb_release.lock* to create the ``pkgb`` package
+again instead of the *pkgb_deps_release.lock* lockfile, it will error, as ``pkgb`` would be already fully
+locked in the former.
```

Comment:
```suggestion
As you can see, the immutability principle remains. If we try to use *pkgb_release.lock* to create the ``pkgb`` package
again instead of the *pkgb_deps_release.lock* lockfile, it will error, as ``pkgb`` would be already fully
locked in the former.
```